full-node/sov-ethereum/src/lib.rs

```diff
@@ -174,16 +182,165 @@ pub mod experimental {
+                        .get_transaction_count(from, None, &mut working_set)
+                        .unwrap_or_default();
+
+                    transaction_request.nonce = Some(reth_primitives::U256::from(nonce.as_u64()));
+                }
+
+                let chain_id = evm
+                    .chain_id(&mut working_set)
+                    .expect("Failed to get chain id")
+                    .map(|id| id.as_u64())
+                    .unwrap_or(1);
+
+                // TODO: implement gas logic after gas estimation is implemented
+                let transaction_request = match transaction_request.into_typed_request() {
+                    Some(TypedTransactionRequest::Legacy(mut m)) => {
+                        m.chain_id = Some(chain_id);
+
+                        TypedTransactionRequest::Legacy(m)
+                    }
+                    Some(TypedTransactionRequest::EIP2930(mut m)) => {
+                        m.chain_id = chain_id;
+
+                        TypedTransactionRequest::EIP2930(m)
+                    }
+                    Some(TypedTransactionRequest::EIP1559(mut m)) => {
+                        m.chain_id = chain_id;
+
+                        TypedTransactionRequest::EIP1559(m)
+                    }
+                    None => {
+                        return Err(to_jsonrpsee_error_object(
+                            "Conflicting fee fields",
+                            ETH_RPC_ERROR,
+                        ));
+                    }
+                };
+
+                let signed_tx = ethereum
+                    .eth_rpc_config
+                    .eth_signer
+                    .sign_transaction(into_transaction(transaction_request), from)
+                    .map_err(|e| to_jsonrpsee_error_object(e, ETH_RPC_ERROR))?;
+
+                RlpEvmTransaction {
+                    rlp: signed_tx.envelope_encoded().to_vec(),
+                }
+            };
+            let (tx_hash, raw_tx) = ethereum
+                .make_raw_tx(raw_evm_tx)
+                .map_err(|e| to_jsonrpsee_error_object(e, ETH_RPC_ERROR))?;
+            let blob = ethereum
+                .batch_builder
+                .lock()
+                .unwrap()
+                .add_transactions_and_get_next_blob(
+                    ethereum.eth_rpc_config.min_blob_size,
+                    vec![raw_tx],
+                );
+            if !blob.is_empty() {
+                ethereum
+                    .submit_batch(blob)
+                    .await
+                    .map_err(|e| to_jsonrpsee_error_object(e, ETH_RPC_ERROR))?;
+            }
+
+            Ok::<_, ErrorObjectOwned>(tx_hash)
         })?;
 
         Ok(())
     }
+
+    // Temporary solution until https://github.com/paradigmxyz/reth/issues/4704 is resolved
+    // The problem is having wrong length nonce/gas_limt/value fields in the transaction request
+    fn into_transaction(request: TypedTransactionRequest) -> reth_primitives::Transaction {
```

Comment:
I know it is temporary but let's call it `try_into_transaction`, return `Result` and use `?` instead of all the unwraps below.